File: Chapter 10/Python/section_10_3_resilience_decorators/fragile/fragile_payment_service.py
import requests

"""
THE FRAGILE IMPLEMENTATION:

DESIGN NOTE:
This class represents the "Naive" way of handling external dependencies in C#. 
It treats a remote network call as if it were a reliable, local constant.

ARCHITECTURAL CRITIQUE:
1. TEMPORAL COUPLING: This method is "Locked" to the network. Without a timeout 
    or retry policy, the calling thread is held hostage by FlakyPayments's server response time.

2. ABSTRACTION LEAK: There is no Interface (Port). The business logic is forced 
    to depend directly on this concrete implementation and the 'HttpClient' library, 
    violating the Downward Dependency Rule (Chapter 4).

3. THE HAPPY PATH FALLACY: The code assumes EnsureSuccessStatusCode() is a 
    finality. In a distributed system, a 503 (Service Unavailable) is often a 
    temporary "hiccup" that could be solved by a retry. Here, it is a fatal crash.
"""

class FragilePaymentService:
    def charge_credit_card(self, amount: float):
        # This raw function is an exposed liability. 
        # It assumes the network is reliable (The Happy Path Fallacy).
        print(f"      [Payment Service] Attempting to charge ${amount}...")

        # If FlakyPayments is down, this line throws a ConnectionError or Timeout.
        # Since there is no retry logic, the user gets an error and the 
        # business loses money.
        response = requests.post(
            "https://api.flakypayments.com/charge", 
            json={"amount": amount, "order_id": "12345"},
            timeout=2 # Even with a timeout, failure is absolute here.
        )
        
        response.raise_for_status() 
        return response.json()

File: Chapter 10/JS/section_10_3_resilience_decorators/fragile/fragilePaymentService.js
/**
 * THE FRAGILE IMPLEMENTATION:
 * * @description
 * This class represents the "Naive" way of handling external dependencies in Node.js. 
 * It treats a remote network call as if it were a reliable, local constant.
 *
 * ARCHITECTURAL CRITIQUE:
 * 1. EVENT LOOP BLOCKING: While 'fetch' is asynchronous, the lack of a timeout 
 * means this Promise could remain pending indefinitely. In Node.js, enough 
 * "hanging" requests will eventually exhaust memory and degrade the 
 * responsiveness of the entire event loop.
 * * 2. ABSTRACTION LEAK: There is no Port (Interface). The business logic is forced 
 * to depend directly on this concrete implementation, violating the 
 * Downward Dependency Rule (Chapter 4).
 * * 3. THE HAPPY PATH FALLACY: The code assumes the network is a "Perfect Pipe." 
 * It provides no mechanism to survive a 503 (Service Unavailable) or a 
 * temporary network "blink." Failure here is absolute and results in a 
 * crashed transaction.
 */
export class FragilePaymentService {
    /**
     * A raw function sitting right in the middle of your business logic.
     * It assumes the outside world is stable and friendly.
     * * @param {number} amount - The dollar amount to charge.
     * @returns {Promise<Object>} The raw JSON response from the vendor.
     */
    async chargeCreditCard(amount) {
        console.log(`      [Payment Service] Attempting to charge $${amount}...`);

        // Raw network call with zero failure policy. 
        // This is an "Unprotected Port."
        const response = await fetch("https://api.flakypayments.com/charge", {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({
                amount: amount,
                order_id: "12345"
            })
            // NOTE: No Signal/AbortController used. We wait forever.
        });

        // Failure is absolute. No second chances.
        if (!response.ok) {
            throw new Error(`Critical failure from FlakyPayments API: ${response.status}`);
        }

        return await response.json();
    }
}